usr/bin/cylc

```diff
@@ -16,54 +16,132 @@
-# 
-# 3) If $CYLC_HOME and $CYLC_VERSION are both undefined, the default
-#   under $CYLC_HOME_ROOT will be selected.
+# If USERS set CYLC_VERSION in their .bashrc it default to an existing value to
+# avoid overiding task job version selection:
+# - export CYLC_VERSION=${CYLC_VERSION:-8.0.0}
+#
+# INSTALLING CYLC RELEASES WITH CONDA
+#------------------------------------
+# Create environments named cylc-$CYLC_VERSION or cylc-flow-$CYLC_VERSION under
+# a root location such as /opt:
+#   $ conda create -p /opt/cylc-8.2.0 cylc=8.2.0
+# and create a default version symlink "cylc" to the latest version:
+#   $ ls /opt/cylc*
+#   cylc-8.1.0
+#   cylc-8.2.0
+#   cylc
+#
+# Note you can `pip install` cylc-flow from a git clone into a Conda
+# environment and then select it by CYLC_VERSION but that will be misleading
+# if the code is not the true release version. It may be better to use Python
+# venvs and CYLC_HOME for cylc-flow development and testing.
+#
+# INSTALLING CYLC-FLOW WORKING COPIES WITH PIP FOR DEVELOPMENT AND TESTING
+#-------------------------------------------------------------------------
+# Create Python virtual environments in-place in your cylc-flow git
+# clones and select them using CYLC_HOME in your .bashrc.
+#
+# Create a new venv and install cylc-flow:
+#   $ cd <my-cylc-flow-clone>
+#   $ python -m venv venv
+#   $ . venv/bin/activate
+#   $ pip install -e . 
+#
+# Set CYLC_HOME in .bashrc to select this venv (see explanation above):
+#   $ echo "export CYLC_HOME=$PWD/venv" >> ~/.bashrc
+#   $ . ~/.bashrc
+#
+# Note you can also `pip install` cylc-flow releases under a common root
+# location for CYLC_VERSION based selection, like for Conda:
+#   $ CYLC_HOME_ROOT=$HOME/cylc-flow/releases  # (set in this wrapper too)
+#   $ python -m venv $CYLC_HOME_ROOT/cylc-flow-8.0a2
+#   $ . $CYLC_HOME_ROOT/cylc-flow-8.0a2/bin/activate
+#   $ python -m pip install cylc-flow==8.0a2
+# But there is no need to do this if you have full system releases - not just
+# cylc-flow - installed via Conda anyway.
  
-#_____________________________
-# NOTES:
-# * Users will typically accept the default or set $CYLC_VERSION, but
-#   they may set $CYLC_HOME to pick up a private cylc installation.
-# * Developers may set $CYLC_HOME to pick up their own repository clone.
-# * Running suites export $CYLC_VERSION to task environments, so that
-#    task messaging commands etc. pick up the compatible cylc version.
-# * Note that $CYLC_ROOT_HOME can be overridden in the environment too.
+##############################!!! EDIT ME !!!##################################
+# Central Cylc Conda releases:
+CYLC_HOME_ROOT="${CYLC_HOME_ROOT:-/opt}"
+# User Conda releases:
+CYLC_HOME_ROOT_ALT="${CYLC_HOME_ROOT_ALT:-$HOME/miniconda3/envs}"
 
-CYLC_HOME_ROOT=${CYLC_HOME_ROOT:-/opt} # !!! EDIT ME !!!
+# Legacy Cylc 7 locations if needed:
+CYLC7_HOME_ROOT="${CYLC7_HOME_ROOT_ALT:-/opt}" 
+CYLC7_HOME_ROOT_ALT="${CYLC7_HOME_ROOT_ALT:-$HOME/cylc/cylc-7}" 
```

Comment:
I'd have thought just having CYLC_HOME_ROOT (defaults to /opt) and CYLC_HOME_ROOT_ALT (no default) would be enough to cover most cases. The logic treats them all the same anyway.

Reply:
I was trying to handle user conda envs out of the box, with _ALT.  But I suppose a different variable name would be more appropriate for that.  (Thinking about it...)

Reply:
I was also allowing for Cylc 7 and 8 to be installed in different locations, but on reflection I suppose that is not necessary. Simplifying...